.github/workflows/mvn-verify-check.yml

```diff
@@ -30,7 +30,45 @@ env:
     -Dskip
 
 jobs:
+  cache-dependencies:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '8', '11', '17' ]
+    # steps:
+    #   - uses: actions/checkout@v3 # refs/pull/:prNumber/merge
+
+    #   - name: Cache local Maven repository
+    #     uses: actions/cache@v3
+    #     with:
+    #       path: ~/.m2/repository
+    #       key: ${{ runner.os }}-maven-${{ hashFiles('**/**/pom.xml') }}
+    #       restore-keys: |
+    #         ${{ runner.os }}-maven-
+    steps:
+      - uses: actions/checkout@v3 # refs/pull/:prNumber/merge
```

Comment:
My understanding every job in this workflow will need to use the cache action, otherwise they don't 

We need to check if this applies to us https://stackoverflow.com/a/71895573  to grab the dependencies downloaded by the first job.

Reply:
We may need to generate a key with the pom of `sql-plugin-api` to use cached deps in `.m2` repo, let me do this and try again.

Reply:
> Error:  Failed to execute goal on project rapids-4-spark-sql-plugin-api_2.12: Could not resolve dependencies for project com.nvidia:rapids-4-spark-sql-plugin-api_2.12:jar:24.02.0-SNAPSHOT: com.nvidia:rapids-4-spark-cdh-bom:pom:24.02.0-SNAPSHOT was not found in https://oss.sonatype.org/content/repositories/snapshots during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of snapshots-repo has elapsed or updates are forced -> [Help 1]

Back to the problem when we add reties. Snapshot cannot be found before executing subsequent jobs to build it. 
Actually, these dependencies from sonatype are most likely to fail the download, because sonatype would loss connection sometimes. I believe the sonatype problem has been well solved by retries, latest failed runs are not related to the problem. 

On the other hand, for speedup purpose, I'm not sure if we can skip the snapshot dependencies, and only pre-download other dependencies.

Reply:
Sorry, I did not notice that these commands fail for Cloudera's dependencies such as -Dbuildver=330cdh.  

these should work

```bash
mvn help:evaluate -pl dist -Dexpression=included_buildvers -DforceStdout -Dmaven.repo.local=$PWD/.m2 -PnoSnapshots -q | tr -d ',' |  xargs -n 1 bash -c 'mvn initialize -pl sql-plugin-api -am -Dbuildver=$1' _

mvn help:evaluate -pl dist -Dexpression=included_buildvers -DforceStdout -Dmaven.repo.local=$PWD/.m2 -PnoSnapshots -q | tr -d ',' |  xargs -n 1 bash -c 'mvn initialize -pl sql-plugin-api -am -Dbuildver=$1' _
```

This should be quite effective for caching because Cloudera shims actually contributed quite a few deps to the total of 3.3G after this fix.

Reply:
Now it works correctly, and the subsequent `cdh` jobs can skip download process. So our cache works too.